### PR TITLE
fix: UserModel::assignIdentities() always produces a DB query

### PR DESCRIPTION
**Pull request by michalsn**
In the `UserModel` - when we use `withIdentities()` method the query is produced properly. 

The problem occurs in the `assignIdentities()` method, when we're trying to assign identities to the users.
We're working with the `User` entity, and there is a line: 
```
$newIdentities   = $mappedUsers[$userId]->identities;
```

It will call `getIdentities()` method which will produce a query to the DB, since there are no identities assigned yet.

This PR solves this problem.

### CODE REVIEW

**Review by datamweb — APPROVED** [review]
@michalsn thank you!

File: tests/Unit/UserTest.php [hunk 1]
```diff
@@ -87,6 +87,35 @@ public function testModelFindByIdWithIdentities(): void
...
 
+    public function testModelFindAllWithIdentitiesWhereInQuery(): void
+    {
+        fake(UserIdentityModel::class, ['user_id' => $this->user->id, 'type' => 'password']);
+        fake(UserIdentityModel::class, ['user_id' => $this->user->id, 'type' => 'access_token']);
+
+        // Grab the user again, using the model's identity helper
+        $users = model(UserModel::class)->withIdentities()->findAll();
+
+        $identities = 0;
```
Comment (kenjis) [inline tests/Unit/UserTest.php:98]:
```suggestion
        $identities = [];
```

**Comment by kenjis** [comment]
Am I correct in assuming that this PR prevents queries to retrieve unnecessary Identities from being executed?
It's just useless and there has never been a problem with the results returned.

**Comment by kenjis** [comment]
The code looks good.

**Comment by michalsn** [comment]
> Am I correct in assuming that this PR prevents queries to retrieve unnecessary Identities from being executed?
It's just useless and there has never been a problem with the results returned.

There was never a problem with the correct result per se. The problem here was that we were producing n+1 queries for code like this:

```php
$users = model(UserModel::class)->withIdentities()->findAll();
foreach ($users as $user) {
    $user->email;
}
```
That would be just fine, but since we use `withIdentities()` it should not work that way.

**Review by kenjis — APPROVED** [review]

**Comment by datamweb** [comment]
Thanks All!

**Pull request merged**